hypothesis-python/RELEASE.rst

```diff
@@ -0,0 +1,13 @@
+RELEASE_TYPE: minor
+
+This release significantly simplifies Hypothesis's internal logic for data
+generation, by removing a number of heuristics of questionable or unproven
+value.
+
+The results of this change will vary significantly from test to test. Most
+test suites will see significantly faster data generation and lower memory
+usage. The "quality" of the generated data may go up or down depending on your
+particular test suites.
+
+If you see any significant regressions in Hypothesis's ability to find bugs in
+your code as a result of this release, please file an issue to let us know.
```

Comment:
I think we need to explicitly note that these changes disable `target()` - we [explicitly marked it as experimental](https://hypothesis.readthedocs.io/en/latest/changes.html#v4-38-0) so that's fine as long as it's not a *silent* noop.

I'd also consider emitting a HypothesisWarning to the effect that it's currently a noop but we expect it to come back soon.

Reply:
I thought about it, and the idea of doing so made me sad, so I thought "What if we just made targeted property based testing work *better* rather than *worse* in this pull request?" so I did that instead.

The targeting algorithm is a fairly simple hill climbing one, but it seems to work reasonably well!

Reply:
though, err, a few teething troubles as evidenced by the broken build. Watch this space.